FIX constantly displaying time tracker notification

vaniakerer opened the pull request:
https://trello.com/c/RFWCjsBW/66-android-q-the-screen-is-always-turned-on-if-timer-is-running

vaniakerer committed:
FIX constantly displaying time tracker notification

https://trello.com/c/RFWCjsBW/66-android-q-the-screen-is-always-turned-on-if-timer-is-running

vaniakerer committed:
Add correct dealing with doze mode and optimization on particular devices

https://trello.com/c/anVLruUi/62-empty-push-is-received-after-stopping-the-timer-and-killing-the-app

AlexanderKozubets commented on app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt:
```diff
@@ -194,9 +194,16 @@ class HrAppTimeTrackerPresenter(
-                timeTrackerInteractor.startTimer(userReport, totalDayMinutesExcludeCurrent(userReport.date.toCalendar(), userReport.minutes))
-                        .subscribe({ updateReportViewModel(TaskUpdate(it.current, TaskTimerState.RUNNING)) }, this::showError)
+                router?.let {
+                    if (!it.isBatteryOptimizationTurnedOn()) {
```
As I understood, we should handle this in all places where we're trying to start a timer, right?
Does it make sense to move this check to the service into `startTimer` method and throw an Exception if optimization is turned on? We can handle our custom exception right in the base fragment in `handleError` method.

vaniakerer replied:
Agree

AlexanderKozubets commented on app/src/main/res/values/strings.xml:
```diff
@@ -9,6 +9,10 @@
     <string name="message_unknown_exception">Unknown exception</string>
+
+    <!--Battery Optimization-->
+    <string name="battery_optimization_turn_off_message">We have to turn off battery optimization for correct work</string>
```
It sounds like a good explanation for a developer, but we need to rephrase it for a user.
I'll get back soon with some suggestions

AlexanderKozubets replied:
Please change text to “Do you agree to disable battery optimization for correct work of timer?”

vaniakerer committed:
Add understandable message to doze mode dialog

https://trello.com/c/anVLruUi/62-empty-push-is-received-after-stopping-the-timer-and-killing-the-app

vaniakerer committed:
Add todo message for future improvements

https://trello.com/c/anVLruUi/62-empty-push-is-received-after-stopping-the-timer-and-killing-the-app
